### ISSUE TITLE
clang-11: error: unknown argument: '-soname'

### ISSUE THREAD

**Issue by VI-gha** [HDFGroup/hdf5#366]
This error occurs when installing HDF5 with AOCC-2.3 (derived from clang/flang) 
It makes it impossible to enable-shared: how to sort this out?

**Comment by VI-gha** [comment]
@lrknox Can anybody help with that?

**Comment by lrknox** [comment]
@VI-gha do you have logs with the error that give more detailed errors?  Other build details would also be useful.  We'll see if we can set up or find AOCC-2.3 on a system available to us, try to reproduce and find the problem and a fix for it.

**Comment by derobins** [comment]
This is/was a libtool error. The libtool script that is generated at compile time was missing the "-Wl," prefix necessary for linker options, so "-Wl,-soname" gets parsed as -soname, which fails.

As of October 2023, with AOCC 4.1, c/flang works out of the box w/ shared unless you are using an MPI compiler wrapper. We have a hack for that (at least in the Autotools), which will appear shortly.

This issue will be closed when the PR is merged to develop.

**Comment by derobins** [comment]
Fixed w/ #3674

**Comment by scemama** [comment]
Here is a patch for libtool that should fix the problem:
```
$ git diff 1ec8fa28dcb29500d485c136db28315671ec4c3b
diff --git a/m4/libtool.m4 b/m4/libtool.m4
index 79a2451e..df1a7860 100644
--- a/m4/libtool.m4
+++ b/m4/libtool.m4
@@ -4719,6 +4719,12 @@ m4_if([$1], [CXX], [
        _LT_TAGVAR(lt_prog_compiler_pic, $1)='-fPIC'
        _LT_TAGVAR(lt_prog_compiler_static, $1)='-static'
         ;;
+      # ARM flang
+      armflang*)
+        _LT_TAGVAR(lt_prog_compiler_wl, $1)='-Wl,'
+        _LT_TAGVAR(lt_prog_compiler_pic, $1)='-fPIC'
+        _LT_TAGVAR(lt_prog_compiler_static, $1)='-static'
+        ;;
       # Lahey Fortran 8.1.
       lf95*)
        _LT_TAGVAR(lt_prog_compiler_wl, $1)='-Wl,'
```
Maybe as the HDF5 group you will have more influence than me to make the libtool developers include it ;-).
